File: surf-cloud-bukkit/src/main/java/dev/slne/surf/cloud/bukkit/processor/BukkitLifecycleProcessor.java
package dev.slne.surf.cloud.bukkit.processor;

import dev.slne.surf.cloud.api.lifecycle.SurfLifecycle;
import dev.slne.surf.cloud.core.processors.AbstractLifecycleProcessor;
import org.bukkit.plugin.java.JavaPlugin;
import org.springframework.stereotype.Component;

@Component
public class BukkitLifecycleProcessor extends AbstractLifecycleProcessor {

  @Override
  protected Class<?> getProvidingClass(SurfLifecycle lifecycle) {
    return JavaPlugin.getProvidingPlugin(lifecycle.getClass()).getClass();
  }
}

File: surf-cloud-bukkit/src/main/kotlin/dev/slne/surf/cloud/bukkit/processor/BukkitLifecycleProcessor.kt
package dev.slne.surf.cloud.bukkit.processor

import dev.slne.surf.cloud.api.common.lifecycle.SurfLifecycle
import dev.slne.surf.cloud.core.common.processors.AbstractLifecycleProcessor
import org.bukkit.plugin.java.JavaPlugin
import org.springframework.stereotype.Component

@Component
class BukkitLifecycleProcessor : AbstractLifecycleProcessor() {
    override fun getProvidingClass(lifecycle: SurfLifecycle) =
        JavaPlugin.getProvidingPlugin(lifecycle.javaClass).javaClass.kotlin
}
